cmd/clusterctl/client/cluster/ownergraph.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// OwnerGraph contains the OwnerReference relationship between objects.
+type OwnerGraph map[string]OwnerGraphNode
+
+// OwnerGraphNode is a single node linking an ObjectReference to its OwnerReferences.
+type OwnerGraphNode struct {
+	Object corev1.ObjectReference
+	Owners []metav1.OwnerReference
+}
+
+func nodeToOwnerRef(n *node, attributes ownerReferenceAttributes) metav1.OwnerReference {
+	ref := metav1.OwnerReference{
+		Name:       n.identity.Name,
+		APIVersion: n.identity.APIVersion,
+		Kind:       n.identity.Kind,
+		UID:        n.identity.UID,
+	}
+	if attributes.BlockOwnerDeletion != nil {
+		ref.BlockOwnerDeletion = attributes.BlockOwnerDeletion
+	}
+	if attributes.Controller != nil {
+		ref.Controller = attributes.Controller
+	}
+	return ref
+}
+
+// GetOwnerGraph returns a data structure with each CAPI controlled object and its OwnerReferences.
+func GetOwnerGraph(namespace string) (OwnerGraph, error) {
```

Comment:
I think it might be better to restructure this function so it's in the test package, even if that means a little bit of copy-past from clusterctl - opinions?